### PR TITLE
Convert IRC colour codes to minetest colour codes.

### PR DESCRIPTION
**Pull request by red-001**

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -87,7 +87,11 @@ setmetatable(irc, {
 
 dofile(modpath.."/config.lua")
 dofile(modpath.."/messages.lua")
-loadfile(modpath.."/hooks.lua")(ie)
+local chunk, errorval = loadfile(modpath.."/hooks.lua")
+if chunk == nil then
+ error(errorval)
+end
+chunk(ie)
```
Comment (ShadowNinja) [inline init.lua]: This can be simplified to `assert(loadfile(modpath.."/hooks.lua"))(ie)`.

Comment (kaeza) [inline init.lua]: This should be split from this commit, IMHO. It's unrelated to the feature being added.

File: hooks.lua [hunk 2]
```diff
@@ -9,14 +9,47 @@ local b64e = ie.require("mime").b64
...
+	[13] = "pink",
+	[14] = "grey",
+	[15] = "lightgrey",
+	[99] = "#ffffff"
+}
+
+local function convert_colour(text)
+	local lines = text:split("\3", true)
+	for index, line in pairs(lines) do
+		local irc_code = line:match("^(%d+)")
```
Comment (kaeza) [inline hooks.lua]: This block would take e.g. "01foo" at the start of the string as an escape sequence.

File: hooks.lua [hunk 3]
```diff
@@ -9,14 +9,47 @@ local b64e = ie.require("mime").b64
 irc.hooks = {}
 irc.registered_hooks = {}
 
+local colour_codes = {
+	[0] = "white",
+	[1] = "black",
```
Comment (kaeza) [inline hooks.lua]: You only really need the index for [0]. The others are auto-numbered starting at 1.

File: hooks.lua [hunk 4]
```diff
@@ -9,14 +9,47 @@ local b64e = ie.require("mime").b64
...
+	[10] = "teal",
+	[11] = "cyan",
+	[12] = "deepskyblue",
+	[13] = "pink",
+	[14] = "grey",
+	[15] = "lightgrey",
+	[99] = "#ffffff"
+}
+
+local function convert_colour(text)
```
Comment (kaeza) [inline hooks.lua]: Minor nit: `color`.

**Review by kaeza — CHANGES_REQUESTED** [review]

**Comment by sofar** [comment]
This seems worth merging, perhaps @kaeza can fix-on-merge or @red-001 red-001 can refresh this?

**Comment by sofar** [comment]
ping?

**Comment by SmallJoker** [comment]
No activity in .. uhm. 6 years. Closing.

**Pull request closed without merging**